Fix spelling mistake in developer_guide.md

cndaimin opened the pull request:
Signed-off-by: cndaimin <daimin@outlook.com>

cndaimin commented:
/assign @gaocegege

gaocegege commented:
Our test infra has some problems. I will let it rerun the test after it works well.

cndaimin commented:
> Our test infra has some problems. I will let it rerun the test after it works well.

ok, Thanks

cndaimin commented:
/retest

johnugeorge commented:
/retest

gaocegege commented:
@cndaimin We fixed the problem in #885 , please rebase the upstream master then have another try :beers:

johnugeorge commented:
/retest

richardsliu commented:
This PR still needs a rebase for the Golang installation fix.

gaocegege commented:
/lgtm

cndaimin commented:
> @cndaimin We fixed the problem in #885 , please rebase the upstream master then have another try 🍻

I have rebased from master instructed by [this guide](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request), and not quite sure does it work.

gaocegege commented:
@cndaimin Thanks for your time. Commit is merged :tada:
